addons/stock_account/views/product_views.xml

```diff
@@ -15,7 +15,7 @@
         <record id="view_category_property_form" model="ir.ui.view">
             <field name="name">product.category.stock.property.form.inherit</field>
             <field name="model">product.category</field>
-            <field name="inherit_id" ref="account.view_category_property_form"/>
+            <field name="inherit_id" ref="product.product_category_form_view"/>
```

Comment:
You should inherit stock.product_category_form_view_inherit since the group is added there